src/redux/preview.js

```diff
@@ -884,3 +884,17 @@ module.exports.updateProjectThumbnail = (id, blob) => (dispatch => {
         dispatch(module.exports.setFetchStatus('project-thumbnail', module.exports.Status.FETCHED));
     });
 });
+
+module.exports.logProjectView = (id, username) => (dispatch => {
+    dispatch(module.exports.setFetchStatus('project-log-view', module.exports.Status.FETCHING));
+    api({
+        uri: `/users/${username}/projects/${id}/views`,
+        method: 'POST'
```

Comment:
I noticed this doesn't take an authentication token.. hypothetically, could I `post` the endpoint above with many different usernames but the same project ID, to artificially increase the project's number of views? It seems like this should require an authentication token. (Actually, it sounds like it really ought to be at the endpoint `/projects/${id}/views`, since you should never be able to post that anybody *else* has viewed a project (only yourself), and the server can detect who is sending the request by viewing the user associated with the authentication token.)

Reply:
The naming is a bit confusing -- this is (as I understand it) only used for recording that this particular user viewed this particular project.

We still might want to require an auth token, though -- I think you might be right that that would be better.

Re: the endpoint location, we're moving towards endpoint paths that include the current user's username, even though that's redundant (if you're sending an auth token), because that can help us avoid unnecessary database lookups.

Reply:
Good catch @towerofnix. Without authentication, the view will not be recorded appropriately. The username in this URL should be the username of the owner of the project. The URL scheme `/users/{owner}/project/{project id}` is a convention that makes it easier to do certain types of validation on the API.

@benjiwheeler The username looks like it's coming from the currently logged in user, but it should be the owner of the project. And I think we need to add `{withCredentials: true}` for this to function properly. I don't think the endpoint will have given any errors about these things though since it's designed to not return errors.